airflow/models/taskinstance.py

```diff
@@ -215,6 +215,49 @@ def try_number(self, value):
     def next_try_number(self):
         return self._try_number + 1
 
+    @staticmethod
+    @provide_session
+    def find(session=None, dag_id=None, task_id=None, execution_date=None,
+             execution_date_before=None, execution_date_after=None,
+             state=None, state_not_equal=None):
+        """
+        Returns a set of dag runs for the given search criteria.
+
+        :param dag_id: the dag_id to find task instance for
+        :type dag_id: int
+        :param task_id: the task_id to find task instance for
+        :type task_id: int
+        :param execution_date_before: filter on execution date before the provided one
+        :type execution_date_before: datetime.datetime
+        :param execution_date_after: filter on execution date after the provided one
+        :type execution_date_after: datetime.datetime
+        :param state_not_equal: the state of the task instance not to be in the results
+        :type state: airflow.utils.state.State
+        :param state: the state of the task instance
+        :type state: airflow.utils.state.State
+        :param session: database session
+        :type session: sqlalchemy.orm.session.Session
+        """
+        TI = TaskInstance
+
+        query = session.query(TI)
+        if dag_id:
```

Comment:
Shouldn't at least dag_id be required?

Reply:
I guess not, what if the user is simply interested in all the data?

Reply:
"What if"s should not be implemented. Have a concrete use case or don't add the code.